src/interpreter_funcs.rs

```diff
@@ -199,7 +199,58 @@ impl Func for FuncImplSeparateIsolatedMeshes {
         let values = mesh_tools::separate_isolated_meshes(geometry);
 
         // FIXME: This returns a slice of Geometries. Return all of them
-        Value::Geometry(Arc::new(values[0].clone()))
+        let first_value = values
+            .into_iter()
+            .next()
+            .expect("Need at least one geometry");
+
+        Value::Geometry(Arc::new(first_value))
+    }
+}
+
+pub struct FuncImplLoopSubdivision;
+impl Func for FuncImplLoopSubdivision {
+    fn flags(&self) -> FuncFlags {
+        FuncFlags::PURE
+    }
+
+    fn param_info(&self) -> &[ParamInfo] {
+        &[
+            ParamInfo {
+                ty: Ty::Geometry,
+                optional: false,
+            },
+            ParamInfo {
+                ty: Ty::Uint,
+                optional: false,
+            },
+        ]
+    }
+
+    fn return_ty(&self) -> Ty {
+        Ty::Geometry
+    }
+
+    fn call(&self, args: &[Value]) -> Value {
+        let geometry = args[0].unwrap_refcounted_geometry();
+        let iterations = args[1].unwrap_uint();
+        if iterations == 0 {
```

Comment:
Wouldn't it help to check if the geometry is triangulated? We already have a method for that.

Reply:
The question is, what do we do if it is not?

Reply:
i think we might need quads for morphing later on. let's keep the possibility for now. 

in this case i suggest triangulating prior to smoothing

Reply:
Okay, but we don't have triangulation yet. So... we'll have to implement it as soon as we add quads. I'll add an `is_triangulated` assert to the `loop_subdivision` function, so that the caller is responsible for triangulating. The operation will have to do the graceful handling, because the user can input anything.